### PR TITLE
Do not enable colorization if stdout is not a TTY

### PR DESCRIPTION
**Pull request by sometimesfood**
If the standard output is not a TTY -- for example when piping output to a pager such as less or redirecting it into a file -- colorization should not be enabled by default.

### CODE REVIEW

**Comment by chastell** [comment]
I thought we [already do this](https://github.com/troessner/reek/blob/ff168d5867546e1f539429d17274cd583b6bdf09/lib/reek/cli/input.rb#L21-L23) – is there a case where it doesn’t work as expected?

**Comment by mvz** [comment]
@chastell that's stdin, not stdout.

This will probably be a tricky feature to get exactly right :-).

**Comment by chastell** [comment]
Argh, right, sorry. I do think we should turn this off for non-TTYs at least, so @sometimesfood is working on a fix and a test for it (and a temporary RuboCop whitelist for the `CLI::Options` class length…). ;)

**Comment by sometimesfood** [comment]
Added a simple spec and the temporary whitelist. Let me know if this is too much brute force.

File: lib/reek/cli/options.rb [hunk 1]
```diff
@@ -9,11 +9,13 @@ module CLI
     #
     # Parses the command line
     #
+    # rubocop:disable ClassLength
```
Comment (chastell) [inline lib/reek/cli/options.rb]: Can you move it to the [`FIXME` exclusion of `.rubocop.yml`](https://github.com/troessner/reek/blob/17c5fbce761727768c3d35c1bc12aa7abfa66c32/.rubocop.yml#L7-L10), please? We do want to fix it eventually. :)

File: spec/reek/cli/options_spec.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
+require_relative '../../spec_helper'
+
+require_relative '../../../lib/reek/cli/options'
+
+describe Reek::CLI::Options do
+  describe '#initialize' do
+    let(:argv) { [] }
```
Comment (chastell) [inline spec/reek/cli/options_spec.rb]: Can you either inline this `let` or make `CLI::Options.new`’s parameter optional? I understand the need for DRY-ness, but I think this makes the individual tests a bit less readable (you have to jump to the `let` just to see that it’s an empty `Array`).

**Comment by sometimesfood** [comment]
@chastell Sure thing, just changed my branch according to your comments.

**Comment by chastell** [comment]
:+1: – I checked the `io/console` stdlib and it seems there’s no easy way to check the colour support of the terminal (and I doubt we want to shell out just to do `tput colors`). We might find a way with one of the gems for colouring console output, but for now I’ll merge this once Travis gets greener.
